third_party/terraform/data_sources/data_source_google_storage_bucket_object.go

```diff
@@ -0,0 +1,42 @@
+package google
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceGoogleStorageBucketObject() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGoogleStorageBucketObjectRead,
+		Schema: map[string]*schema.Schema{
+			"object": &schema.Schema{
```

Comment:
If both fields are required, then what does using this data source actually get you?

Reply:
AFAIK this is because people want terraform to throw an error when a certain object doesn't exist in cloud storage. One example is people who are using [cloud functions](https://www.terraform.io/docs/providers/google/r/cloudfunctions_function.html) with terraform where the cloud function code is sourced from a cloud storage bucket.

That being said I think I didn't fully understand data sources before starting this. I've taken a second attempt at this.